compiler/rustc_borrowck/src/diagnostics/mod.rs

```diff
@@ -939,7 +939,20 @@ impl<'cx, 'tcx> MirBorrowckCtxt<'cx, 'tcx> {
             return OtherUse(use_span);
         }
 
-        for stmt in &self.body[location.block].statements[location.statement_index + 1..] {
+        // drop and replace might have moved the assignment to the next block
+        let maybe_additional_statement =
+            if let TerminatorKind::Drop { target: drop_target, .. } =
+                self.body[location.block].terminator().kind
+            {
+                Some(self.body[drop_target].statements.iter().take(1))
```

Comment:
```suggestion
                self.body[drop_target].statements.first()
```
Then you can remove the `.into_iter().flatten()` below.